### DISCUSSION TITLE
how to change tooltip offset?

### DISCUSSION THREAD

**Discussion by kfalcami** [zauberzeug/nicegui#3880] [Q&A] [answered]
### Question

According to the Quasar docs, tooltip property "offset" is an array of 2 numbers, default [14, 14].  I'm able to change the anchor and self props (both strings), but can't figure out how to change the offset.  This one the offset is apparently ignored:

```py
ui.tooltip().props('anchor="top middle" self="bottom middle" offset=[100,100]')
```

and this moves the tooltip to the left edge of the browser with various numbers I've tried including the documented default [14, 14]:

```py
ui.tooltip().props('anchor="top middle" self="bottom middle" offset="[100,100]"')
```

Both correctly change the anchor and self props.

**Comment by falkoschindler** [comment, marked as answer]
Hi @kfalcami,

The problem with the string "[100,100]" is that it needs to be converted to a JavaScript array on the client. This can be done using the prefix `:`:
```py
with ui.label('Hello'):
    ui.tooltip('World').props('anchor="top middle" self="bottom middle" :offset="[100,100]"')
```

**Comment by kfalcami** [comment]
Yep, that worked, thanks.  I looked through the docs but couldn't find an example of this -- how would I know which props need the colon?  Everything that's not a string maybe?  

I looked around to see if there was an obvious place in the nicegui docs where I might add something, no luck.  Maybe in one of the ui.tooltip examples?  That would've helped me anyway :)

**Reply by python-and-novella** [reply]
props can't convert string to  javascript object. so , if you want pass non-string in props, you need to add prefix ':'.

**Reply by falkoschindler** [reply]
@kfalcami You're right, this is one of the things that are not easy to find in the documentation. Mentioning in a `ui.tooltip` demo would have helped. But other users coming across this issue with another element wouldn't expect to find it there.

Once in a while I'm thinking about some documentation page about "general principles" with a collection of these kind of things. Maybe even the starting page https://nicegui.io/documentation could serve this purpose if we add some demos here and there, mentioning `.style`, `.classes`, `.props`, `.add_slot`, `.on`, `:`, running CPU- or IO-bound tasks and so on.
